### PR TITLE
Add remaining source documentation for Platypus

### PR DESCRIPTION
**Pull request by alexanderianblair**
Initial source documentation for Platypus was recently reviewed and merged in #43, but documentation for PRs merged during this process (for commits in #37, #41, and #42) was not included. This PR adds documentation for the new classes and example added in those PRs.

### CODE REVIEW

**Review by k-collie — CHANGES_REQUESTED** [review]
Thanks Alex, just a few small changes and me not being sure about the maths.

File: doc/content/source/bcs/MFEMVectorBoundaryIntegratedBC.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+## Overview
+
+Adds the boundary integrator for integrating the linear form
+
+!equation
+(\vec f, \vec v)_{\partial\Omega} \,\,\, \forall \vec v \in V
+
+where $v \in \vec H^1$, $\vec f$ is a vector coefficient of the same dimension, and
+$\hat n$ is the outward facing unit normal vector on the boundary.
```
Comment (cmacmackin) [inline doc/content/source/bcs/MFEMVectorBoundaryIntegratedBC.md:15]: $\hat n$ doesn't actually seem to be used in this file.

File: doc/content/source/kernels/MFEMLinearElasticityKernel.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+!syntax description /Kernels/MFEMLinearElasticityKernel
+
+## Overview
+
+Adds the domain integrator for integrating the bilinear form
+
+!equation
+(\sigma_{ij}, \partial_j v_i)_\Omega \,\,\, \forall v_i \in V
+
+where $v_i \in H^1$, and $\sigma$ is the stress tensor of an material with an isotropic stress/strain
```
Comment (cmacmackin) [inline doc/content/source/kernels/MFEMLinearElasticityKernel.md:14]: "a material"

File: doc/content/examples/LinearElasticity.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
+# Linear Elasticity Problem
+
+## Summary
+
+Solves a 3D linear elasticity problem for
+  the deformation of a mult-material cantiliver beam. This example
+  is based on [MFEM Example 2](https://mfem.org/examples/).
```
[the remark this answers is not in the document]
Comment (alexanderianblair) [reply doc/content/examples/LinearElasticity.md]: They shouldn't be - fixed

File: doc/content/source/actions/SetMeshFESpaceAction.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+Sets the nodal mesh FESpace to use the same FESpace as the
+[`MFEMVariable`](source/variables/MFEMVariable.md) used to describe node displacements.
+
+## Overview
+
+Action called to set the nodal mesh FESpace to use the same FESpace as the variable used to describe
+node displacements in problems involving deformed meshes, parsing content inside the `Mesh` block in
+the user input. Only has an effect if the `Problem` type is set to
+[`MFEMProblem`](source/problem/MFEMProblem.md) and the `Mesh` type is set to [`MFEMMesh`](source/mesh/MFEMMesh.md)
```
[the remark this answers is not in the document]
Comment (alexanderianblair) [reply doc/content/source/actions/SetMeshFESpaceAction.md:13]: Don't think this should have a comma here - both statements need to be true for `SetMeshFESpaceAction` to have this effect.

File: doc/content/source/bcs/MFEMVectorBoundaryIntegratedBC.md (hunk 1 above)
Comment (alexanderianblair) [reply doc/content/source/bcs/MFEMVectorBoundaryIntegratedBC.md]: Indeed - removed

File: doc/content/source/kernels/MFEMLinearElasticityKernel.md (hunk 2 above)
Comment (alexanderianblair) [reply doc/content/source/kernels/MFEMLinearElasticityKernel.md]: Thanks - corrected

File: doc/content/examples/LinearElasticity.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+\begin{split}
+\sigma_{ij} \equiv C_{ijkl} \varepsilon_{kl} = \lambda \delta_{ij} \varepsilon_{kk} + 2 \mu \varepsilon_{ij} \\
+\varepsilon_{ij} = \frac{1}{2} \left(\partial_i u_j + \partial_j u_i\right)
+\end{split}
+\end{equation}
+
+In this example, we solve this using the weak form
+
+\begin{equation}
+(\sigma_{ij}, \partial_j v_i)_\Omega - (\sigma_{ij} \hat n_j, v_i)_{\partial \Omega}
```
[the remark this answers is not in the document]
Comment (alexanderianblair) [reply doc/content/examples/LinearElasticity.md:47]: Pretty sure this is still correct - the first term is just $(\sigma : \nabla v)$ written in index notation.
[eg. see here](https://jsdokken.com/dolfinx-tutorial/chapter2/linearelasticity.html).

The MFEM documentation has mixed index and vector notation there, which I think is where the confusion is coming from.

The second term is indeed zero away from $\Gamma_2$ here, but I'm keen to keep this notation as I find it slightly more descriptive for showing where the BC term arises in the maths,

File: doc/content/source/kernels/MFEMLinearElasticityKernel.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+## Summary
+
+!syntax description /Kernels/MFEMLinearElasticityKernel
+
+## Overview
+
+Adds the domain integrator for integrating the bilinear form
+
+!equation
+(\sigma_{ij}, \partial_j v_i)_\Omega \,\,\, \forall v_i \in V
```
[the remark this answers is not in the document]
Comment (alexanderianblair) [reply doc/content/source/kernels/MFEMLinearElasticityKernel.md:12]: Addressed above

File: doc/content/source/actions/SetMeshFESpaceAction.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (alexanderianblair) [reply doc/content/source/actions/SetMeshFESpaceAction.md:13]: Ah ignore me - missed that! Yes, this makes sense

File: doc/content/examples/LinearElasticity.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+## Description
+
+This problem solves a linear elasticity problem for small deformations with strong form:
+
+\begin{equation}
+\begin{split}
+-\partial_j \sigma_{ij} =0
+\,\,\,&\mathrm{on}\,\, \Omega \\
+\delta u_i = 0 \,\,\, &\mathrm{on}\,\, \Gamma_1 \\
```
[the remark this answers is not in the document]
Comment (alexanderianblair) [reply doc/content/examples/LinearElasticity.md]: We can indeed - done

File: doc/content/examples/LinearElasticity.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (k-collie) [reply doc/content/examples/LinearElasticity.md:47]: Ok I'm happiest with all index notation but I can follow that, I'm more convinced given that bilinear forms should map to a scalar.

**Review by k-collie — APPROVED** [review]

**Review by cmacmackin — APPROVED** [review]

**Pull request merged**